sans/reduction.ipynb

```diff
@@ -210,6 +210,55 @@
     "    return (pixel_size * pixel_length) / (L2 * L2)"
    ]
   },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def q_resolution(lam_edges, moderator, d):\n",
+    "    moderator = sc.rebin(moderator, 'wavelength', lam_edges)\n",
+    "    \n",
+    "    d_lam = lam_edges['wavelength', 1:] - lam_edges['wavelength', :-1] # bin widths\n",
+    "    lam = 0.5 * (lam_edges['wavelength', 1:] + lam_edges['wavelength', :-1]) # bin centres\n",
+    "    \n",
+    "    r2 = sc.Variable(value=4.0824829046386295/1000, unit=sc.units.m) # sample aperture radius\n",
+    "    r1 = sc.Variable(value=14.433756729740645/1000, unit=sc.units.m) # source aperture radius\n",
+    "    dr = sc.Variable(value=8.0/1000, unit=sc.units.m) # virtual ring width on detector\n",
+    "    \n",
+    "    l2 = sc.neutron.l2(d)\n",
+    "    l1 = sc.Variable(value=5.0, unit=sc.units.m) # should be sc.neutron.l1(d) = 25.6m ?\n",
+    "    theta = sc.neutron.scattering_angle(d)  \n",
+    "\n",
+    "    inv_l3 = (l1 + l2) / (l1*l2)\n",
+    "    \n",
+    "    # Terms in Mildner and Carpenter equation. See https://docs.mantidproject.org/nightly/algorithms/TOFSANSResolutionByPixel-v1.html\n",
+    "    a1 = (r1/l1)*(r1/l1) * 3.0\n",
+    "    a2 = (r2*inv_l3)*(r2*inv_l3) * 3.0\n",
+    "    a3 = (dr/l2) * (dr/l2) \n",
+    "    q = 4.0 * np.pi * sc.sin(theta) * sc.reciprocal(lam) # keep in wav dim to prevent broadcast\n",
+    "    \n",
+    "    h = 6.62607004E-34\n",
+    "    m = 1.6749E-27\n",
+    "    k = sc.Variable(value=h/m, unit=(sc.units.m * sc.units.m)/sc.units.s) \n",
+    "    \n",
+    "    moderator.data.values *= 1e-6 # us to seconds\n",
+    "    moderator.data.unit = sc.units.s # fixup moderator units to microseconds. Previously dimensionless\n",
```

Comment:
Is this an bug in our Mantid converter?

Reply:
I'll take a look at what the original outputs were. I think they might have been dimensionless in the `LoadRKH` generated workspace, but if not correct units could be attached in the converter.